feat(studio): update query performance table ui

kemaldotearth opened the pull request:
## I have read the [CONTRIBUTING.md](https://github.com/supabase/supabase/blob/master/CONTRIBUTING.md) file.

YES

## What kind of change does this PR introduce?

This updates the Query Performance table to use some of our newer UI and fix things that aren't currently optimal. Changes introduced are as follows:

- Column header styling brushed up.
- Column headers now sortable as data grids elsewhere are.
- Numbers aligned left for legibility.
- Improved syntax highlighting and how it caps in the cell.
- Cells given minimum widths.
- Columns now ordered by more relevant stats first.
- Fixed Index Advisor popover not being accessible. 
- Fixed total time returning string instead of number.

| Before | After |
|--------|--------|
| <img width="1195" height="785" alt="Screenshot 2025-09-08 at 21 48 09" src="https://github.com/user-attachments/assets/1fb91180-d9e2-4d4d-92fb-95109f359b11" /> | <img width="1196" height="792" alt="Screenshot 2025-09-08 at 21 47 36" src="https://github.com/user-attachments/assets/be3f3cee-f9a1-418d-9287-5371aa6f1d5d" /> |

kemaldotearth committed:
feat: tidy up query perf column headers

kemaldotearth committed:
feat: update col widths and which ones are sortable

This gives a reasonable column width to all cols as well as adds an option for us to filter out which columns we dont want to make the table sortable by.

kemaldotearth committed:
feat: column order and word wrapping for query

kemaldotearth committed:
feat: column data text align for numbers

kemaldotearth committed:
fix: pointer event was preventing index advisor icon popover

kemaldotearth committed:
feat: use code block instead of editor

Swapped Editor out for Code block as its less expensive and modified it to add pgsql syntax highlighting

kemaldotearth committed:
fix: small truncation issue

kemaldotearth committed:
Merge remote-tracking branch 'origin/master' into feat/query-perf-table-update

joshenlim committed:
Merge branch 'master' into feat/query-perf-table-update

kemaldotearth committed:
fix: merge conflict with imports

kemaldotearth committed:
Merge branch 'feat/query-perf-table-update' of github.com:supabase/supabase into feat/query-perf-table-update

kemaldotearth committed:
chore: remove unused sort function

joshenlim commented on apps/studio/components/interfaces/QueryPerformance/QueryPerformanceGrid.tsx:
```diff
@@ -54,24 +53,66 @@ export const QueryPerformanceGrid = ({ queryPerformanceQuery }: QueryPerformance
+                  <DropdownMenuItem
+                    onClick={() => {
+                      setSort({ column: col.id, order: 'asc' })
+                      router.push({
+                        ...router,
+                        query: { ...router.query, sort: col.id, order: 'asc' },
```
reckon we use `nuqs` to handle URL state

joshenlim commented on apps/studio/components/interfaces/QueryPerformance/QueryPerformanceGrid.tsx:
```diff
@@ -54,24 +53,66 @@ export const QueryPerformanceGrid = ({ queryPerformanceQuery }: QueryPerformance
       name: col.name,
       resizable: true,
       minWidth: col.minWidth ?? 120,
+      sortable: !nonSortableColumns.includes(col.id),
```
if we're introducing sorting - reckon we should probably also save the sort order in local storage, otherwise the sort order will keep getting reset on each tab session

joshenlim replied:
btw this might not be working - im trying on the staging preview but im not able to sort the columns by dragging them

kemaldotearth committed:
fix: sort menu closing on first click

Sort menu of a column no longer disappears on first click

kemaldotearth committed:
feat: move sorting and ordering over to nuqs

This removes the usage of Next Router params and uses Nuqs instead for managing the QP table sorting and ordering.

kemaldotearth committed:
Merge remote-tracking branch 'origin/master' into feat/query-perf-table-update

kemaldotearth committed:
Merge remote-tracking branch 'origin/master' into feat/query-perf-table-update

kemaldotearth committed:
feat: add clear sorting button to filter bar

kemaldotearth committed:
chore: remove unused files

kemaldotearth committed:
fix: add highlighting for selected sort

kemaldotearth commented on apps/studio/components/interfaces/QueryPerformance/QueryPerformanceGrid.tsx:
```diff
@@ -27,51 +29,77 @@ import {
-              <>{sort.order === 'desc' ? <ArrowDown size={14} /> : <ArrowUp size={14} />}</>
+
+            {isSortable && (
+              <DropdownMenu>
```
Not sure Select applies here as we can't change the icon? Took inspiration from our Table Editor. I do agree it needs something, so I've highlighted the sorting that's selected in the dropdown. Same effect more or less. 

<img width="483" height="198" alt="Screenshot 2025-09-09 at 18 18 59" src="https://github.com/user-attachments/assets/0bd23e65-0bb1-490f-ae0a-089d0e7b2fb8" />

joshenlim replied:
yeah we probably don't need the icons tbh - but can revisit this again, prioritise on consistency across the dashboard how we'd wanna handle sort dropdowns like these

dnywh commented on apps/studio/components/interfaces/QueryPerformance/QueryPerformanceGrid.tsx:
```diff
@@ -27,51 +29,83 @@ import {
+                    )}
+                  >
+                    <ArrowUp size={14} />
+                    Sort Ascending
```
Minor nit: make sentence case like other buttons/actions? I.e. `Sort ascending`. Ditto for what’s currently `Sort Descending`

dnywh replied:
Hmm, I see this elsewhere like on our current sorting for Table Editor

dnywh commented on apps/studio/components/interfaces/QueryPerformance/QueryPerformanceFilterBar.tsx:
```diff
@@ -98,6 +101,12 @@ export const QueryPerformanceFilterBar = ({
             onSaveFilters={onFilterRolesChange}
             className="w-56"
           />
+
+          {sort && (
+            <Button size="tiny" type="link" icon={<X size={12} />} onClick={clearSort}>
+              Clear sort
+            </Button>
+          )}
```
I don’t see any visual feedback on _how_ the table is currently sorted. Should that be here?

joshenlim committed:
Nit update sort

kemaldotearth committed:
nit: smol styling update for direction

kemaldotearth committed:
Merge remote-tracking branch 'origin/master' into feat/query-perf-table-update

kemaldotearth committed:
feat: inian feedback on time based numbers

kemaldotearth committed:
fix: return number instead of string for total time

We were returning a string for total time percentage which meant the sorting was a little bit off

kemaldotearth committed:
Merge remote-tracking branch 'origin/master' into feat/query-perf-table-update

joshenlim committed:
Shift roles to nuqs

kemaldotearth committed:
Merge remote-tracking branch 'origin/master' into feat/query-perf-table-update

joshenlim reviewed (approved):
Verified changes in staging preview!

kemaldotearth committed:
Merge remote-tracking branch 'origin/master' into feat/query-perf-table-update

kemaldotearth committed:
Merge remote-tracking branch 'origin/master' into feat/query-perf-table-update

kemaldotearth committed:
chore: clean up

kemaldotearth committed:
Merge remote-tracking branch 'origin/master' into feat/query-perf-table-update

fsansalvadore commented on apps/studio/components/interfaces/QueryPerformance/QueryPerformanceFilterBar.tsx:
```diff
@@ -98,6 +100,20 @@ export const QueryPerformanceFilterBar = ({
+              </p>
+              <Tooltip>
+                <TooltipTrigger onClick={clearSort}>
+                  <X size={14} className="text-foreground-light hover:text-foreground" />
```
@kemaldotearth noticed the text in this element gets stacked if viewport too narrow.

<img width="121" height="69" alt="Screenshot 2025-09-10 at 18 06 08" src="https://github.com/user-attachments/assets/b9eea205-032c-4d63-9ad1-7b69371224ad" />

Might want to avoid it going on multiple lines? 
And maybe that whole actions bar could be horizontally scrollable if the width exceeds the available space?

fsansalvadore replied:
another component where I used the horizontal scroll for responsiveness for reference:

https://github.com/supabase/supabase/blob/master/apps/studio/components/layouts/ProjectLayout/LayoutHeader/LayoutHeader.tsx#L131C1-L132C1

https://github.com/user-attachments/assets/b9c08f92-3b81-472b-b369-fc2cb8d210a9

fsansalvadore reviewed (approved):
Approving as everything looks good overall, just left one comment @kemaldotearth

kemaldotearth committed:
fix: smaller viewport filter bar
